docs/enhancements/streamlined-templates.md

```diff
@@ -0,0 +1,214 @@
+# Streamlined Templates
+
+## How it works now
+
+Currently, Sonobuoy plugins are large, sprawling YAML files. This works, but it
+makes the boundary between Sonobuoy and the plugin very messy.
+
+By way of illustration, here is the current `heptio-e2e` plugin file. Every part
+of user configuration is in green, all boilerplate required by Sonobuoy is in
+red.
+
+```diff
+- ---
+- apiVersion: v1
+- kind: Pod
+- metadata:
+-   annotations:
++     sonobuoy-driver: Job
++     sonobuoy-plugin: heptio-e2e
++     sonobuoy-result-type: heptio-e2e
+-   labels:
+-     component: sonobuoy
+-     sonobuoy-run: '{{.SessionID}}'
+-     tier: analysis
+-   name: sonobuoy-heptio-e2e-job-{{.SessionID}}
+-   namespace: '{{.Namespace}}'
+- spec:
+-   containers:
++   - image: gcr.io/heptio-images/heptio-e2e:master
++     imagePullPolicy: Always
++     name: heptio-e2e
+-     volumeMounts:
+-     - mountPath: /tmp/results
+-       name: results
+-       readOnly: false
+-   - command:
+-     - sh
+-     - -c
+-     - /sonobuoy worker global -v 5 --logtostderr
+-     env:
+-     - name: NODE_NAME
+-       valueFrom:
+-         fieldRef:
+-           fieldPath: spec.nodeName
+-     - name: RESULTS_DIR
+-       value: /tmp/results
+-     - name: MASTER_URL
+-       value: '{{.MasterAddress}}'
+-     - name: RESULT_TYPE
+-       value: heptio-e2e
+-     image: gcr.io/heptio-images/sonobuoy:master
+-     imagePullPolicy: Always
+-     name: sonobuoy-worker
+-     volumeMounts:
+-     - mountPath: /tmp/results
+-       name: results
+-       readOnly: false
+-   restartPolicy: Never
+-   serviceAccountName: sonobuoy-serviceaccount
+-   tolerations:
+-   - effect: NoSchedule
+-     key: node-role.kubernetes.io/master
+-     operator: Exists
+-   - key: CriticalAddonsOnly
+-     operator: Exists
+-   volumes:
+-   - emptyDir: {}
+-     name: results
+```
+
+## Summary
+
+Removing this boiler plate will not just greatly simplify the process of making
+a plugin. It will also allow us to make changes behind the scenes to the
+Sonobuoy execution model without requiring updates to template files.
+
+All this boiler plate isn't just annoying. It paralyses changes to the Sonobuoy
+execution model. Adding
+
+## Objectives
+
+Replace the existing templates with a greatly streamlined one.
+
+### Goals
+* Conceal the execution mechanics from users
+* Allow seamless upgrading of Sonobuoy without user intervention
+
+### Non-goals
+* Do not modify the done contract betweens producer and consumer container
+
+## Proposals
+
+### Proposal 1
+
+An entirely new YAML-based file will be used for plugins. This will consist of
+two sections: Metadata and container spec.
+
+```yaml
+sonobuoy-config:
+  driver: Job
+  plugin-name: heptio-e2e
+  result-type: heptio-e2e
+spec:
+  image: gcr.io/heptio-images/heptio-e2e:master
+    imagePullPolicy: Always
+    name: heptio-e2e
+    volumeMounts:
+    - mountPath: /tmp/results
```

Comment:
I like it, we just need to make certain it's clear that the producer is expected to output to the location of the producer.